### PR TITLE
Adding BambooHR extension

### PR DESCRIPTION
**Pull request by RobErskine**
## Description

Added a new extension to interface with the [BambooHR](https://www.bamboohr.com/) [API](https://documentation.bamboohr.com/docs). This extension allows users to tap into some of the most popular HR features like:
- **Employee Directory** - Search all employees in your organization to find their photo, name, pronouns, department, location, manager, and more.
- **Who's Out Today?** - Get a list of everyone who is out of the office today.
- **Upcoming PTO** - Review your PTO requests for you and your direct reports, including their dates and approval status
- **Calculator PTO** - Review all the different categories that your organization offers and how much time you have taken to date, as well as how much you have remaining. 

_Sneaking in a minor change in the RescueTime Focus Trigger extension_
- I previously worked on this extension and added a small update interval to it so the command updates in the background every few minutes

## Screencast

Screencast demo using the command and the rationale for it:

https://github.com/raycast/extensions/assets/1895120/ea6040a8-3195-4c0a-b9fe-0ae4436254f0

Forgot to include preferences in my video:
BambooHR Preferences

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
npm run lint && npm run build passing

- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by RobErskine** [comment]
Sorry, if sneaking in a small one-liner to the RescueTime command messes everything up for adding a net-new Command, I can close this and submit two separate PRs.

**Comment by pernielsentikaer** [comment]
Regarding the update to the rescue time extension, only no-view and menubar accept interval 🫠

You can see [it here](https://developers.raycast.com/information/manifest#command-properties)

Can you move it from the PR so we don't break something 🙂

**Comment by RobErskine** [comment]
hey @pernielsentikaer i want to be respective of the upcoming holidays so you and the rest of the Raycast team can enjoy some well-deserved time off. Should I close and re-open in the new year? If there is anything I can do to help, please let me know! Cheers 🍻

File: extensions/bamboohr/src/utils/types.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export interface Preferences {
+  bamboo_api_key: string;
+  bamboo_subdomain: string;
+  bamboo_user_id: string;
+}
+
```
Comment (pernielsentikaer) [inline extensions/bamboohr/src/utils/types.ts:6]: You already have them in `ExtensionPreferences` within raycast-env.d.ts

```suggestion

```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and have a single feedback ready for you 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by pernielsentikaer** [comment]
Maybe just like this? 🙂

**Comment by RobErskine** [comment]
@pernielsentikaer apologies, sorry I missed your [Update commit](https://github.com/raycast/extensions/pull/9684/commits/2b6bb443bad9a5177ed99eb216f129f95403e9b7). I just pulled it down and was successfully able to run the production build as well as continue to dev locally; everything works great!

**Review by RobErskine — COMMENTED** [review]
@pernielsentikaer apologies, sorry I missed your [Update commit](https://github.com/raycast/extensions/pull/9684/commits/2b6bb443bad9a5177ed99eb216f129f95403e9b7). I just pulled it down and was successfully able to run the production build as well as continue to dev locally; everything works great!

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**